File: CineBox/CineBox.Services/Database/Screening.cs
using System;
using System.Collections.Generic;

namespace CineBox.Services.Database;

public partial class Screening
{
    public int Id { get; set; }

    public int MovieId { get; set; }

    public int HallId { get; set; }

    public string Category { get; set; } = null!;

    public DateTime ScreeningTime { get; set; }

    public decimal Price { get; set; }

    public virtual ICollection<Booking> Bookings { get; set; } = new List<Booking>();

    public virtual Hall Hall { get; set; } = null!;

    public virtual Movie Movie { get; set; } = null!;
}

File: CineBox/UI/cinebox_desktop/lib/models/Screening/screening.dart
import 'package:cinebox_desktop/models/Movie/movie.dart';
import 'package:json_annotation/json_annotation.dart';

part 'screening.g.dart';

@JsonSerializable()
class Screening {
  int? id;
  int? movieId;
  int? hallId;
  String? category;
  DateTime? screeningTime;
  double? price;
  Movie? movie;

  Screening(
      this.id, this.movieId, this.hallId, this.category, this.screeningTime, this.movie);

  /// A necessary factory constructor for creating a new User instance
  /// from a map. Pass the map to the generated `_$UserFromJson()` constructor.
  /// The constructor is named after the source class, in this case, User.
  factory Screening.fromJson(Map<String, dynamic> json) =>
      _$ScreeningFromJson(json);

  /// `toJson` is the convention for a class to declare support for serialization
  /// to JSON. The implementation simply calls the private, generated
  /// helper method `_$UserToJson`.
  Map<String, dynamic> toJson() => _$ScreeningToJson(this);
}
